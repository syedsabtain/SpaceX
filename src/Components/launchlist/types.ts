export interface Fnc{

    handleIdChange: (newid:number)=>void
}