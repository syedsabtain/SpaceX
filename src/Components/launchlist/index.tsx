import React from 'react'
import {useLaunchListQuery} from '../../generated/graphql'
import Launchlist from './launchlist'
import {Fnc} from '../launchlist/types'



const List : React.FC<Fnc> = ({handleIdChange}) => {
    const {data, loading, error} = useLaunchListQuery();

    if (loading) {
        return (
            <div className="d-flex justify-content-center text-white mt-5">
                <div className="spinner-border" role="status">
                    <span className="sr-only">Loading...</span>
                </div>
            </div>
        )
    }

    if (error || !data) {
        return (
            <h1>{error}</h1>
        )
    }
   
    const props ={ data:data ,handleIdChange:handleIdChange }
    return (
        <div className='row'>
            <div className='col-md-12'>
            <Launchlist {... props}></Launchlist>
            </div>
        </div>
    )
}

export default List