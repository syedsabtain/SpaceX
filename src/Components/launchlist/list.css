.custom-li{
    background-color: rgba(5, 5, 5, 0.507)!important;
    /* border-radius: 5px; */
    /* padding: 10px; */
    color: rgb(255, 255, 255)!important;
    text-shadow: 2px 2px 4px #000000!important;
    
}
.only{
    background-color: rgba(5, 5, 5, 0)!important;
    /* border-radius: 5px; */
    /* padding: 10px; */
    color: rgb(255, 255, 255)!important;
    border-width: 0px!important;
    text-shadow: 2px 2px 4px #000000!important;
}
.custom-ul{
    /* height: 10%;
    overflow-y: scroll; */
   
    
}
.custom-ul button{
    transition: background-color 0.5s;
    transition: font-size 0.5s;
}
.custom-ul button:hover{

    background-color: rgba(5, 5, 5, 0.712)!important;
     font-size: 20px;
    
     
}