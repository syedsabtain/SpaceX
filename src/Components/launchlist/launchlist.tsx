import React from 'react'
import {LaunchListQuery} from '../../generated/graphql'
import {Fnc} from  './types'
import './list.css'
interface Props extends Fnc {

    data : LaunchListQuery;
}

const Launchlist : React.FC < Props > = ({data,handleIdChange}) => {

    return (
        <div className="list-group bg-transparent custom-ul">
            <h4 className="list-group-item mt-2 text-center  only" >Mission List</h4>
            {data.launches?.map((value,key)=>{
                return(
                    <button key={key} type="button" className="list-group-item  custom-li " onClick={()=>{ handleIdChange(value?.flight_number!)}} >
                    {value?.mission_name} ({value?.launch_year})
              </button>
                )
            })}
        </div>
    )
}

export default Launchlist