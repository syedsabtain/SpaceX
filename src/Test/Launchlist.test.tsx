import React from 'react';
import { shallow, mount } from 'enzyme';
import Launchlist from '../Components/launchlist/launchlist';
import { Fnc } from '../Components/launchlist/types';

describe('Launchlist Component',()=>{

    it('div',()=>{
        
        const props = { data:{},handleIdChange:()=>{}}
        const com = shallow(<Launchlist {... props}></Launchlist>)
        expect(com.find('div').length).toEqual(1);

    })
    it('h4',()=>{
        const props = { data:{},handleIdChange:()=>{}}
        const com = shallow(<Launchlist {... props}></Launchlist>)
        expect(com.find('h4').length).toEqual(1)
    })
});

test('renders the component', () => {

    const props = { data:{},handleIdChange:()=>{}}
        const com = shallow(<Launchlist {... props}></Launchlist>)
    expect(com).toMatchSnapshot();
  
  
  });