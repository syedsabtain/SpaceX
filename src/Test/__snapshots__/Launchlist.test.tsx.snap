// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<div
  className="list-group bg-transparent custom-ul"
>
  <h4
    className="list-group-item mt-2 text-center  only"
  >
    Mission List
  </h4>
</div>
`;
